templates/kafka/0/docker-compose.yml

```diff
@@ -0,0 +1,16 @@
+zk:
```

Comment:
The changes you described vs our ZK sound like they would be fine to me as a PR, but I'll let @cloudnautique review the specifics.  I have no issue with "Kafka should have its own dedicated instance of ZK".  The part I don't like is really not related to you specifically at all, but about services including separate dependent services in general.. sharing the definition of a service, not the deployed instances of it:

- Including it changes from "should have a dedicated" to "you **will** have a dedicated".. maybe I have a good reason to want to share with something else.

- I'm now tightly coupled to your version of ZK. I can't use the Rancher one, or someone-elses-even-awsomer-zookeeper template, or an External Service that points to the ZK-as-a-service my company runs, or a 3rd party one hosted elsewhere, or my own that has an extra option for a knob I need to tweak that you didn't think of.

- As the Kafka maintainer you now have to keep up to date with ZK releases and update the Kafka template every time there's a new one instead of letting someone who maybe cares about and keeps up with ZK more often do it.  As a user I have to worry about whether your version of ZK packaged in is up to date.

The alternative is you have a question with `type: service`, `description: "Choose the Zookeeper service to use.  It should really be dedicated just to Kafka, see $url for more info."`.  The main disadvantage to me is really that it's less convenient.. the user has to have already created one.  And also that it then shows up as a separate service even though it's (presumably) dedicated to Kafka.

I think we really need a couple more features to do something like nest catalog services and have a question for "choose an existing or deploy a new $service" that would allow you to reference ZK without including a new copy of it..

Anyway if you feel strongly it has to be a dedicated ZK then this is the only way to enforce that and we can merge this as-is.  But I would really prefer separating it out improving the Rancher zookeeper and/or including your own rawminds-awesome-zookeper here.